### PR TITLE
Feature : Data Persistence on Local Storage Data

### PR DESCRIPTION
**Pull request by ich-wtag**
**Overview**
This PR includes data persistence using Local Storage as well as **Filter State** also saved to sync with the State change on UI.

**Types Of Change**

- [x] **Data** and **Filter State** saved in the **Local Storage**
- [x] While refreshing **Data** and **Filter State** are taken from **Local Storage** and updated present data and state on **UI**
- [x]  The **UI** of the before and after refreshing is synced.

[Local-storage-Data-persistence.webm](https://github.com/ich-wtag/Todo-App_Vanilla_Js/assets/153156568/126bd82a-4096-4fe0-9d8e-d6a1d559d0c1)

### CODE REVIEW

File: js/script.js [hunk 1]
```diff
@@ -369,11 +375,27 @@ const renderTodos = () => {
 
     activateFilterButton(todos, filterState);
 
+    updateLocalStorageData();
```
Comment (hhu-wtag) [inline js/script.js]: User applied search, filter and pagination and refreshed the screen. 
Will this logic persist all the state?

Comment (ich-wtag) [reply js/script.js]: No bhai, only filter state. So do I have to make all the state persistent?

Comment (ich-wtag) [reply js/script.js]: If it is I can implement thats too.

Comment (hhu-wtag) [reply js/script.js]: Yes. Please

File: js/script.js [hunk 2]
```diff
@@ -369,11 +386,39 @@ const renderTodos = () => {
...
+    updateLocalStorageData(searchedValue);
+
     paginatedTodos.forEach((todo) => {
         $todoList.appendChild(createTodoElement(todo));
     });
 };
 
+const getLocalStorageData = () => {
+    const data = JSON.parse(localStorage.getItem("todos"));
+    const state = JSON.parse(localStorage.getItem("state"));
```
Comment (hhu-wtag) [inline js/script.js]: `JSON.parse`, `JSON.stringfy` are prone to break. Move them to a separate file and use `try..catch` block.

```js
// dbOperations.js

export function getTodos() {
  const payload = {
    todos: [],
    error: null
  }
  try {
    payload.todos = JSON.parse(localStorage.getItem("todos"))
  } catch (error) {
    payload.error = error
  } finally {
    return payload
  }
}
```

**Usage:**
```js
const {todos, error} = getTodos()

if(error){
  // handle error here by showing toast notification
  return
}

const {state, error} = getState()

if(error){
  return
}

// Do work with todos and state
```

**Review by hhu-wtag — APPROVED** [review]

**Pull request merged**